hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -764,8 +790,18 @@ public AbfsInputStream openFileForRead(Path path,
       String resourceType, eTag;
       long contentLength;
       EncryptionAdapter encryptionAdapter = null;
-      if (fileStatus instanceof VersionedFileStatus
-          && client.getEncryptionType() != EncryptionType.ENCRYPTION_CONTEXT) {
+      /*
+      * GetPathStatus API has to be called in case of:
+      *   1.  fileStatus is null or not an object of VersionedFileStatus: as eTag
+      *       would not be there in the fileStatus object.
+      *   2.  fileStatus is an object of VersionedFileStatus and the object doesn't
+      *       have encryptionContext field when client's encryptionType is
+      *       ENCRYPTION_CONTEXT.
+      * */
+      if ((fileStatus instanceof VersionedFileStatus) && (
```

Comment:
In-line with https://github.com/apache/hadoop/pull/3440#discussion_r989075168. Here the expectation is that the:
1. Fetch List of VersionFileStatus objects by listStatus API of the AzureBlobFileSystem.
2. Use the fetchedStatus to be sent as an option while doing openWithOptions().

Tested in https://github.com/pranavsaxena-microsoft/hadoop/blob/HADOOP-17912-PR-Review-2/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsCustomEncryption.java#L273-L297.